service/templates/cloudformation/auto_scaling_group.yaml

```diff
@@ -12,6 +12,10 @@
       HealthCheckGracePeriod: {{ .HealthCheckGracePeriod }}
       MetricsCollection:
         - Granularity: "1Minute"
+      Tags:
+        - Key: Name
+          Value: {{ .ClusterID }}-{{ .ASGType }}
+          PropagateAtLaunch: true
```

Comment:
For the tags we need to have KubernetesCluster. This is used by the cloud provider for persistent volumes and load balancers. We should use the same tags as the current ASG.

https://github.com/giantswarm/aws-operator/blob/afdb61b9c5f3ccbbab67423c2ede52a1b17b814a/resources/aws/auto_scaling_group.go#L83

Reply:
Thanks! Sone, in a local test the stack was created, the api was responding but no nodes showed up, could it be related to this?

Reply:
It could be but it might also be a problem registering with the Ingress ELB. Is it set correctly in the ASG?

Reply:
Sorry it's not that I can see the instances are registered OK. Plus that would just affect ingress and not them appearing as nodes.

I don't think it's the tag because the cloud provider is optional. Can you SSH to the workers using the VPC peering?

Reply:
Nope, from master it asks for password

Reply:
Sounds like a problem with the cloud config and there are no SSH users.

The only way around that I've found is re-enabling the EC2 keypair in the legacy resource and providing it to the ASG.